src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EarlyExitManager.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class EarlyExitManager
+    {
+        private IBlockFinder? _blockFinder;
```

Comment:
you managed to extract classes with separated responsibility and it seems a right approach but you missed an opportunity to test them in isolation -> both here and in the GasPriceOracle